indexer-api/src/domain/storage/transaction.rs

```diff
@@ -66,4 +66,7 @@ where
         &self,
         session_id: SessionId,
     ) -> Result<(Option<u64>, Option<u64>, Option<u64>), sqlx::Error>;
+
+    /// Get the highest end_index from all transactions (for progress calculation).
+    async fn get_highest_transaction_end_index(&self) -> Result<Option<u64>, sqlx::Error>;
```

Comment:
I think we already have this baked into `get_highest_indices`; there it returns this value plus others needed for the wallet subscription (should we rename it relevant_transaction_subscription?). Could we avoid duplication without sacrificing performance (`get_highest_indices` is a single query)? Probably not, so this should be fine; just wanted you to think about it.

Reply:
Actually this would belong into `TransactionStorage`. But with my other comment below, this might need to go away and be folded into `UnshieldedUxtoStorage::get_highest_end_index_for_address`.